### ISSUE TITLE
When deployed (not using emulators) the login expires too fast

### ISSUE THREAD

**Issue by silviot** [eisbuk/EisBuk#99]
It seems like after one hour the user is being kicked out.

It smells like our app never refreshes its token.

See https://firebase.google.com/docs/auth/admin/manage-sessions

**Comment by silviot** [comment]
I tried enabling the [Security Token Service API](https://console.cloud.google.com/marketplace/product/google/sts.googleapis.com?q=search&referrer=search&project=eisbuk) following a recommendation [I found here](https://groups.google.com/g/firebase-talk/c/MR4YAyTpIWs?pli=1).

**Comment by silviot** [comment]
Ok, I think I got to the bottom of this!
[This SO question helped a lot!](https://stackoverflow.com/questions/48164248/bad-requests-in-token-service-api-when-using-firebase-auth).

I was being a bit paranoid about what the API key allowed you to do, since we need to ship it with the web app and is thus semi-public. So I went to remove as many services as possible that it allowed access to. Turns out one of these services was the one that allowed tokens to be refreshed. With that service disabled users could still log in, but when their token expires they'll receive an access denied error when trying to refresh it.

Now that I enabled the _Token Service API_ besides the _Identity Toolkit API_ it's working again.
